audio-worklet/src/noisegate-script-processor.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+class NoiseGate {
+  /**
+   * A Noise gate allows audio signals to pass only when the registered volume
+   * is above a specified threshold.
+   * @param {BaseAudioContext} context the audio context
+   * @param {Number} options.channels the number of input and output
+   *                                  channels, a maximum of two
+   * @param {Object} options parameters for the noise gate
+   * @param {Number} options.attack seconds for gate to fully close
+   * @param {Number} options.release seconds for gate to fully open
+   * @param {Number} options.bufferSize the size of an onaudioprocess window
+   * @param {Number} options.timeConstant seconds for envelope follower's
+   *                                   smoothing filter alpha weight
+   * @param {Number} options.threshold decibel level beneath which sound is
+   *                                   muted
+   */
+  constructor(context, options) {
+    if (!(context instanceof BaseAudioContext))
+      throw 'Not a valid audio context.';
```

Comment:
Have you checked this message? If you get this exception in the middle of a large console log, you'll have no clue where it is from. Perhaps this is what you'll see:

```js
Uncaught Not a valid audio context
```

You don't have to fix it here, but it's good to know.